storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -103,3 +103,52 @@ func (d *DotGit) Idxfile() (fs.FS, string, error) {
 
 	return nil, "", ErrIdxNotFound
 }
+
+func (d *DotGit) Objectfiles() (fs.FS, []core.Hash, error) {
+	dotGitobjcts := d.fs.Join(d.path, "objects")
```

Comment:
There is an 'e' missing from this variable name and the second 'o' must be in uppercase:

dotGitobjcts -> dotGitObjects

Maybe just rename it to objsDir instead, it will be more coherent with the rest of the code in this file.